packages/core/src/invoke.ts

```diff
@@ -1,163 +1,166 @@
 import { MachineNode } from './MachineNode';
 
 export const DEFAULT_SPAWN_OPTIONS = { sync: false };
 
+export function machineToActor(
+  machine: MachineNode<any, any>,
+  id: string,
+  parent: Actor,
+  options?: { sync?: boolean }
+): Actor {
+  const service = interpret(machine, {
+    ...options, // inherit options from this interpreter
+    parent,
+    id
+  });
+
+  if (options && options.sync) {
+    service.onTransition(state => {
+      parent.send({
+        type: actionTypes.update,
+        state,
+        id: service.id
+      });
+    });
+  }
+
+  service
+    .onDone(doneEvent => {
+      parent.send(
+        toSCXMLEvent(doneInvoke(service.id, doneEvent.data), {
+          origin: service.id
+        })
+      );
+    })
+    .start();
+
+  return service as Actor;
+}
 export function spawnMachine<
   TContext,
   TEvent extends EventObject,
   TMachine extends MachineNode<any, any, any>
 >(
   machine: TMachine | ((ctx: TContext, event: TEvent) => TMachine),
   options: { sync?: boolean } = {}
-): InvokeCreator<TContext, TEvent> {
+): ActorCreator<TContext, TEvent> {
   return (ctx, event, { parent, id, data, _event }) => {
     let resolvedMachine = isFunction(machine) ? machine(ctx, event) : machine;
     if (data) {
       resolvedMachine = resolvedMachine.withContext(
         mapContext(data, ctx, _event)
       ) as TMachine;
     }
-    const childService = interpret(resolvedMachine, {
-      ...options, // inherit options from this interpreter
-      parent,
-      id: id || resolvedMachine.id
-    });
 
-    const resolvedOptions = {
-      ...DEFAULT_SPAWN_OPTIONS,
-      ...options
-    };
+    return machineToActor(resolvedMachine, id, parent, { sync: options.sync });
+  };
+}
 
-    if (resolvedOptions.sync) {
-      childService.onTransition(state => {
-        parent.send({
-          type: actionTypes.update,
-          state,
-          id: childService.id
-        });
-      });
-    }
+export function promiseToActor<T>(
+  promise: PromiseLike<T>,
+  id: string,
+  parent: Actor
+): Actor {
+  let canceled = false;
 
-    childService
-      .onDone(doneEvent => {
+  promise.then(
+    response => {
+      if (!canceled) {
         parent.send(
-          toSCXMLEvent(doneInvoke(id, doneEvent.data), {
-            origin: childService.id
-          })
+          toSCXMLEvent(doneInvoke(id, response) as any, { origin: id })
         );
-      })
-      .start();
+      }
+    },
+    errorData => {
+      if (!canceled) {
+        const errorEvent = error(id, errorData);
+        try {
+          // Send "error.platform.id" to this (parent).
+          parent.send(toSCXMLEvent(errorEvent as any, { origin: id }));
+        } catch (error) {
+          reportUnhandledExceptionOnInvocation(errorData, error, id);
+        }
+      }
+    }
+  );
 
-    const actor = childService;
+  const actor = {
+    id,
+    send: () => void 0,
+    subscribe: (next, handleError, complete) => {
```

Comment:
`subscribe` from all actor types can be dropped, because you have removed this from the `Actor` interface. Because those "converters" only return Actor type those subscribe methods are "lost" anyway - the calling/consuming code cant use them because its not defined on the Actor interface